code/harness/DiskContents.cpp

```diff
@@ -0,0 +1,298 @@
+#include "DiskContents.h"
+using std::endl;
+
+namespace fs_testing {
+
+fileAttributes::fileAttributes() {
+  dir_attr = NULL;
+  stat_attr = NULL;
+  md5sum = "";
+}
+
+fileAttributes::~fileAttributes() {
+}
+
+void fileAttributes::set_dir_attr(struct dirent* a) {
+  dir_attr = (struct dirent *) malloc(sizeof(struct dirent));
+  dir_attr->d_ino = a->d_ino;
+  dir_attr->d_off = a->d_off;
+  dir_attr->d_reclen = a->d_reclen;
+  dir_attr->d_type = a->d_type;
+  strncpy(dir_attr->d_name, a->d_name, sizeof(a->d_name));
+  dir_attr->d_name[sizeof(a->d_name) - 1] = '\0';
+}
+
+void fileAttributes::set_stat_attr(struct stat* a) {
+  stat_attr = (struct stat *) malloc(sizeof(struct stat));
+
+  stat_attr->st_dev = a->st_dev;
+  stat_attr->st_ino = a->st_ino;
+  stat_attr->st_mode = a->st_mode;
+  stat_attr->st_nlink = a->st_nlink;
+  stat_attr->st_uid = a->st_uid;
+  stat_attr->st_gid = a->st_gid;
+  stat_attr->st_rdev = a->st_rdev;
+  stat_attr->st_size = a->st_size;
+  stat_attr->st_atime = a->st_atime;
+  stat_attr->st_mtime = a->st_mtime;
+  stat_attr->st_ctime = a->st_ctime;
+  stat_attr->st_blksize = a->st_blksize;
+  stat_attr->st_blocks = a->st_blocks;
+}
+
+void fileAttributes::set_md5sum(std::string file_path) {
+  FILE *fp;
+  std::string command = "md5sum " + file_path;
+  char md5[100];
```

Comment:
Can you guarantee that this will always be big enough to get all the contents you need? What if an error occurs or the file path and the file path is long? You could do something like [what Tester.cpp has](https://github.com/utsaslab/crashmonkey/blob/master/code/harness/Tester.cpp#L530)